client/src/components/HostMenu.jsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+import io from 'socket.io-client';
+import Dashboard from './Dashboard.jsx';
+
+const url = 'http://localhost:8080';
+const socket = io(url);
+
+const HostMenu = ({ guests, menu, setTopLevelState }) => {
+  const [currentName, setCurrentName] = useState('');
+
+  useEffect(() => {
+    socket.on('orderSubmitted', payload => {
+      console.log('Orders: ', payload);
+      const guestArray = [...guests, payload];
+      setTopLevelState('guests', guestArray);
+    });
+  });
+
+  const onChange = (event) => {
+    setCurrentName(event.target.value);
+    onCloseClick();
+    event.preventDefault();
+  };
+
+  const addItem = (item) => {
+    const guestArray = [...guests];
+    for (let i = 0; i < guestArray.length; i++) {
+      if (guestArray[i].guestName === currentName) {
+        if (guestArray[i].order.find(menuItem => menuItem.name === item.name)) {
+          alert('Item already added');
+          return;
+        }
+        guestArray[i].order.push(item);
+        setTopLevelState('guests', guestArray);
+        return;
+      }
+    }
+  };
+
+  const menuItems = menu.menu_sections.map((section, i) => {
+    return (
+      <details key={i}>
+        <summary> {section.section_name} </summary>
+        <ul>
+        {section.menu_items.map((item, j) => {
+          return (
+            <li key={j}>
+              {item.name} ${item.price}
+              <button type='button' onClick={() => { addItem(item); }}> + </button>
```

Comment:
Same as in one of my earlier comments re: the `{}` after `=>`

Reply:
I left this as-is since removing the {} brackets just returns `addItem(item)` instead of calling it.

Reply:
Wait nm my comment doesn't make sense. Let me check this out